hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -401,6 +427,8 @@ private static void modify(FileSystem fs, Path metadataFolder, Properties modify
           throw new HoodieIOException("Checksum property missing or does not match.");
         }
       }
+
+      // 5. delete the backup properties file
       fs.delete(backupCfgPath, false);
```

Comment:
Optionally, we could remove the delete of the backupfile, so that the previous version of hoodie.properties would be available as backup.  Helps with debugging, on what properties changed recently.